### PR TITLE
preserve gene metadata column order when reading/writing anndata

### PR DESCRIPTION
**Pull request by schlemmge**
io.py renamed to inout.py due to pycharm issue with this specific file name

### CODE REVIEW

File: fastgenomics/external/anndata.py [hunk 1]
```diff
@@ -71,6 +81,8 @@ def read_gene_metadata(gene_meta, expr):
         check_file_type(gene_meta, content_type="gene_metadata")
         df = pd.read_csv(gene_meta.path, dtype=GENE_METADATA_DTYPE)
 
+    original_column_order = [str(_) for _ in df.columns]
```
Comment (ChristianSauer) [inline fastgenomics/external/anndata.py]: df.columns.to_list()

File: tests/test_external.py [hunk 2]
```diff
@@ -96,3 +96,87 @@ def test_read_write(fgprocess_1, adata_1):
...
+    )
+
+    # the annotated column order
+    assert adata_1.uns[fgad.GENE_METADATA_COLUMN_ORDER] is not None
+
+    written_gene_metadata_columns = pd.read_csv(
+        fgprocess_1.output["genes_output"].path
+    ).columns
+
+    print("WRITTEN")
```
Comment (ChristianSauer) [inline tests/test_external.py]: remove?

File: tests/test_external.py [hunk 3]
```diff
@@ -96,3 +96,87 @@ def test_read_write(fgprocess_1, adata_1):
...
+    for i in written_gene_metadata_columns:
+        print(i)
+
+    print("ORIGINAL")
```
Comment (ChristianSauer) [inline tests/test_external.py]: remove?

File: tests/test_external.py [hunk 4]
```diff
@@ -96,3 +96,87 @@ def test_read_write(fgprocess_1, adata_1):
...
+    for i in adata_1.uns[fgad.GENE_METADATA_COLUMN_ORDER]:
+        print(i)
```
Comment (ChristianSauer) [inline tests/test_external.py]: whats the point of the test? should this be an assert?=

File: fastgenomics/external/anndata.py [hunk 5]
```diff
@@ -125,4 +138,10 @@ def write(adata, expr=None, cell_meta=None, gene_meta=None):
 
     if gene_meta is not None:
         check_file_type(gene_meta, content_type="gene_metadata")
-        adata.var.to_csv(gene_meta.path)
+        if GENE_METADATA_COLUMN_ORDER in adata.uns.keys():
+            adata.var.reset_index().\
```
Comment (ChristianSauer) [inline fastgenomics/external/anndata.py]: Kann das unerwartete Nebeneffekte haben? Z.b. wenn jemand anders einen index setzt? Wahrscheinlich irrelevant

**Comment by pwl** [comment]
Closing in favor of #39

**Pull request closed without merging**